multiqc/modules/mirtrace/mirtrace.py

```diff
@@ -0,0 +1,381 @@
+            'ylab': '# Reads',
+            'cpswitch_counts_label': 'Number of Reads'
+        }
+
+        return bargraph.plot(data, keys, config)
+
+
+    # miRTrace Read Length Distribution
+    def mirtrace_length_plot(self):
+        """ Generate the miRTrace Read Length Distribution"""
+
+        data = dict()
+        for s_name in self.length_data:
+            try:
+                data[s_name] = {int(d): int(self.length_data[s_name][d]) for d in self.length_data[s_name]}
+            except KeyError:
+                pass
+        if len(data) == 0:
+            log.debug('No valid data for read length distribution')
+            return None
+
+        config = {
+            'id': 'mirtrace_length_plot',
+            'title': 'miRTrace: Read Length Distribution',
+            'ylab': 'Read Count',
+            'xlab': 'Read Lenth (bp)',
+            'ymin': 0,
+            'xmin': 0,
+            'xDecimals': False,
+            'tt_label': '<b>Read Length (bp) {point.x}</b>: {point.y} Read Count',
+            'xPlotBands': [
+                {'from': 40, 'to': 50, 'color': '#ffebd1'},
+                {'from': 26, 'to': 40, 'color': '#e2f5ff'},
+                {'from': 18, 'to': 26, 'color': '#e5fce0'},
+                {'from': 0, 'to': 18, 'color': '#ffffe2'},
+            ]
+        }
+
+        return linegraph.plot(data, config)
+
+
+    # miRTrace RNA Categories
+    def mirtrace_rna_categories(self):
+        """ Generate the miRTrace RNA Categories"""
+
+        # Specify the order of the different possible categories
+        keys = OrderedDict()
+        keys['reads_mirna']      =   { 'color': '#33a02c', 'name': 'miRNA' }
+        keys['reads_rrna']       =   { 'color': '#ff7f00', 'name': 'rRNA' }
+        keys['reads_trna']       =   { 'color': '#1f78b4', 'name': 'tRNA' }
+        keys['reads_artifact']   =   { 'color': '#fb9a99', 'name': 'Artifact' }
+        keys['reads_unknown']    =   { 'color': '#d9d9d9', 'name': 'Unknown' }
+
+        # Construct a data structure for the plot - duplicate the samples for read 1 and read 2
+        data = {}
+        for s_name in self.summary_data:
+            data[s_name]                                =    {}
+            data[s_name]['reads_mirna']      =   self.summary_data[s_name]['reads_mirna']
+            data[s_name]['reads_rrna']       =   self.summary_data[s_name]['reads_rrna']
+            data[s_name]['reads_trna']       =   self.summary_data[s_name]['reads_trna']
+            data[s_name]['reads_artifact']   =   self.summary_data[s_name]['reads_artifact']
+            data[s_name]['reads_unknown']    =   self.summary_data[s_name]['reads_unknown']
+
+
+        # Config for the plot
+        config = {
+            'id': 'mirtrace_rna_categories_plot',
+            'title': 'miRTrace: RNA Categories',
+            'ylab': '# Reads',
+            'cpswitch_counts_label': 'Number of Reads'
+        }
+
+        return bargraph.plot(data, keys, config)
```

Comment:
As above, you can just use `self.summary_data` directly instead of creating `data`